File: DSA-DAILY/01-08-2025/118. Pascal's Triangle@leetcode-easy.py
from typing import List
class Solution:
    def generate(self, num: int) -> List[List[int]]:
        dp=[[1]]
        for i in range(2,num+1):
            dp.append([1]+[dp[-1][j]+dp[-1][j+1]for j in range(i-2)]+[1])
        return dp
if __name__ == "__main__":
    sol=Solution()
    n=int(input("Enter the number of rows for Pascal's Triangle:"))
    res=sol.generate(n)
    for row in res:
        print(row)

File: DSA-DAILY/01-08-2025/118. Pascal's Triangle@leetcode-easy.js
/**
 * @param {number} num
 * @return {number[][]}
 */
const generate = num => {
    const dp=[[1]];
    for(let i=1;i<num;i++){
        const prev=dp[dp.length-1];
        const paddedleft= [0, ...prev];
        const paddedright=[...prev,0];
        const curr=paddedleft.map((val,ind) => val + paddedright[ind]);
        dp.push(curr);
    }
    return dp;
};

const readline=require('readline');
const rl=readline.createInterface({
    input: process.stdin,
    output: process.stdout.out
});

rl.question('Enter the number of rows for Pascal\'s Triangle: ', input => {
    const n=parseInt(input);
    const res=generate(n);
    console.log('Pascal\'s Triangle:',res);
    rl.close();
});

const main=()=>{
    const n=readInput();
    const Triangle=generate(n);
    console.log(Triangle);
    Triangle.forEach(row => console.log(row.join(' ')));
}